lib/appdmg.js

```diff
@@ -10,24 +10,38 @@ var util = require('./util');
 var hdiutil = require('./hdiutil');
 var Pipeline = require('./pipeline');
 
-module.exports = exports = function (source, target, cb) {
+module.exports = exports = function (config, cb) {
 
   if (process.platform !== 'darwin') {
     throw new Error('Platform not supported: ' + process.platform);
   }
 
-  var resolveBase = path.dirname(source);
+  var basepath = config.basepath;
+  var specification = config.specification;
+  var source = config.source;
+  var target = config.target;
+
+  if (!basepath) {
+    if (source) {
+      basepath = path.dirname(source);
+    } else {
+      basepath = process.cwd();
+    }
+  }
+
   var resolvePath = function (to) {
-    return path.resolve(resolveBase, to);
+    return path.resolve(basepath, to);
   };
 
   var global = {
-    source: source,
     target: target
   };
 
-  delete source;
-  delete target;
```

Comment:
Don't delete variables: http://jslinterrors.com/only-properties-should-be-deleted